### ISSUE TITLE
OpenXR local floor emulation is not working on HTC Elite (and possibly other devices)

### ISSUE THREAD

**Issue by BastiaanOlij** [godotengine/godot#93647]
### Tested versions

- Reproducible in 4.3 95110ddcb41ba4b9b1f8c9bf58c8910f6616e60a

### System information

Android - Compatibility

### Issue description

We recently introduce support for the local floor reference space and this is working perfectly on systems that support this mode (Quest, SteamVR). Trying this out on an HTC Elite XR it didn't work properly, the camera gets positioned on the floor. My initial conclusion is that local floor is not supported and the emulation isn't functioning. This likely applies to other platforms as well.

It is possible that in recent refactors we broke the emulation logic.

Note, this is not a showstopper, local floor can also be emulated with stage and using `XRServer.center_on_hmd` on a pose recentre request.

### Steps to reproduce

- Use the MRP
- Install the vendor plugin
- Configure export template for Khronos plugin
- Deploy to HTC Elite

### Minimal reproduction project (MRP)

Use the demo project introduced here: https://github.com/godotengine/godot-demo-projects/pull/1080

**Comment by BastiaanOlij** [comment]
cc @dsnopek

**Comment by dsnopek** [comment]
I'm able to reproduce this at startup on the HTC Vive XR Elite. However, when I use the system re-center, it does place me at the correct height, which leads me to believe it is sort of work, just not at startup.

> It is possible that in recent refactors we broke the emulation logic.

When first implementing the emulation, I tested it on the Quest by hard-coding the check for `XR_EXT_local_floor` to false.

I suspect we broke the emulation, in a similar way to how we briefly broke composition layers (fixed in PR https://github.com/godotengine/godot/pull/91686) due to the play space being available a little later, but I still have to dig into it
